provider/oauth2/views.py

```diff
@@ -137,3 +143,19 @@ def invalidate_access_token(self, at):
         else:
             at.expires = now() - timedelta(days=1)
             at.save()
+
+
+class AccessTokenDetailView(View):
+    def get(self, request, *args, **kwargs):
+        JSON_CONTENT_TYPE = 'application/json'
+
+        try:
+            access_token = AccessToken.objects.get_token(kwargs['token'])
+            content = {
+                'username': access_token.user.username,
+                'scope': access_token.get_scope_display(),
+                'expires_in': access_token.get_expire_delta()
+            }
+            return HttpResponse(json.dumps(content), content_type=JSON_CONTENT_TYPE)
```

Comment:
I was curious about the explicit JSON response in this view, is this because this auth project isn't using DRF?